### ISSUE TITLE
False positive on JDK 27

### ISSUE THREAD

**Issue by mpkorstanje** [uber/NullAway#1534]
After upgrading `error_prone_core` to v2.49.0 and `nullaway` to v0.13.3 together (due to https://github.com/uber/NullAway/issues/1511) in https://github.com/junit-team/junit-framework/pull/5589, https://github.com/junit-team/junit-framework/actions/runs/24429467026/job/71370536336?pr=5589 fails on  https://github.com/junit-team/junit-framework/pull/5589/changes/bc2ffaced372fc63a2f1f6e1547a214d9a3cb753 with a false positive.

```java
@Override
	public void close() {
		if (!this.closed) {
			try {
				if (this.closeAction != null) {
					List<Throwable> failures = new ArrayList<>();
					this.storedValues.entrySet().stream() //
							.map(e -> EvaluatedValue.createSafely(e.getKey(), e.getValue())) //
							.filter(Objects::nonNull) //
							.sorted(EvaluatedValue.REVERSE_INSERT_ORDER) //
							.forEach(it -> {
								try {
									it.close(this.closeAction);
								}
								catch (Throwable t) {
									UnrecoverableExceptions.rethrowIfUnrecoverable(t);
									failures.add(t);
								}
							});
					if (!failures.isEmpty()) {
						var iterator = failures.iterator();
						var throwable = iterator.next();
						iterator.forEachRemaining(throwable::addSuppressed);
						throw throwAsUncheckedException(throwable);
					}
				}
			}
			finally {
				this.closed = true;
			}
		}
	}
```

With:

```
home/runner/work/junit-framework/junit-framework/junit-platform-engine/src/main/java/org/junit/platform/engine/support/store/NamespacedHierarchicalStore.java:146: warning: [NullAway] dereferenced expression it is @Nullable

									it.close(this.closeAction);
									  ^
    (see http://t.uber.com/nullaway )
```

Here `it` can not be `null` because `.filter(Objects::nonNull)` was applied.  The same false positive doesn't show up on JDK 26 and below.

Probably immediately detected once https://github.com/uber/NullAway/issues/1525 is resolved, but I didn't want to keep this from you either.

**Comment by msridhar** [comment]
Thanks for the report!  I think I tracked down the root cause.  Aiming to have a fix soon.

**Comment by msridhar** [comment]
With #1540 and #1541 I think JUnit builds successfully on JDK 27 ea+17, so I'm going to close.  We'll cut a new release soon with these fixes.

**Comment by msridhar** [comment]
@mpkorstanje NullAway 0.13.4 is out now and should have the relevant fixes

**Comment by mpkorstanje** [comment]
Yes! Build fails to fail, as expected. Cheers!

https://github.com/junit-team/junit-framework/actions/runs/24738870333/job/72373061724?pr=5598
